### PR TITLE
[BEAM-2640, BEAM-2641] Introduces TextIO.read().withHintMatchesManyFiles()

### PR DESCRIPTION
**Pull request by jkff**
In that case it expands to TextIO.readAll(). Implementing this when the filepattern is a ValueProvider nudged me to also implement BEAM-2640 - Create.ofProvider(ValueProvider).

Links:
https://issues.apache.org/jira/browse/BEAM-2640
https://issues.apache.org/jira/browse/BEAM-2641

R: @reuvenlax 
CC: @sammcveety

### CODE REVIEW

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Create.java [hunk 1]
```diff
@@ -485,6 +496,44 @@ protected boolean advanceImpl() throws IOException {
...
 
+  /** Implementation of {@link #ofProvider}. */
+  public static class OfValueProvider<T> extends PTransform<PBegin, PCollection<T>> {
+    private final ValueProvider<T> provider;
+    private final Coder<T> coder;
+
+    private OfValueProvider(ValueProvider<T> provider, Coder<T> coder) {
+      checkNotNull(provider, "provider");
+      checkNotNull(coder, "coder");
+      this.provider = provider;
```
Comment (reuvenlax) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Create.java]: this.provider = checkNotNull(provider, "provider");
this.coder = checkNotNull(coder, "coder");

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Create.java [hunk 2]
```diff
@@ -485,6 +496,44 @@ protected boolean advanceImpl() throws IOException {
...
+      if (provider.isAccessible()) {
+        Values<T> values = Create.of(provider.get());
+        return input.apply(values.withCoder(coder));
+      }
+
+      @SuppressWarnings({"unchecked", "rawtypes"})
+      SerializableCoder<ValueProvider<T>> providerCoder =
+          SerializableCoder.of((Class) provider.getClass());
+      return input
+          .apply(Create.of(provider).withCoder(providerCoder))
```
Comment (reuvenlax) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Create.java]: why do you need this? The ValueProvider is already serialized in the transform. You can simply Create.of((Void) null) and then expand in the next ParDo.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Create.java (hunk 1 above)
Comment (jkff) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Create.java]: Done

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Create.java (hunk 2 above)
Comment (jkff) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Create.java]: Done

**Comment by jkff** [comment]
Ping.

**Comment by jkff** [comment]
This PR got mistakenly closed - that commit should have actually closed https://github.com/apache/beam/pull/3599 instead.

**Comment by jkff** [comment]
Reincarnated this PR in https://github.com/apache/beam/pull/3639

**Comment by kennknowles** [comment]
Sorry for the typo! It wasn't even needed, since it was a merge commit.

**Pull request closed without merging**